redwood-treehouse-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/treehouse/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -40,50 +50,76 @@ internal class UIViewLazyList<A : AppService>(
   treehouseApp: TreehouseApp<A>,
   widgetSystem: WidgetSystem,
 ) : LazyList<UIView> {
-  private val dataSource = TableViewDataSource(treehouseApp, widgetSystem)
-  private val root = UITableView().apply {
-    this.dataSource = this@UIViewLazyList.dataSource
-  }
+  private val layout = UICollectionViewFlowLayout()
+  private val viewController = CollectionViewController(treehouseApp, widgetSystem, layout)
 
   override fun isVertical(isVertical: Boolean) {
-    if (!isVertical) {
-      // TODO UITableView only supports vertical scrolling. Switch to UICollectionView.
-      TODO()
+    layout.scrollDirection = if (isVertical) {
+      UICollectionViewScrollDirectionVertical
+    } else {
+      UICollectionViewScrollDirectionHorizontal
     }
   }
 
   override fun intervals(intervals: List<LazyListInterval>) {
-    dataSource.intervals = intervals
-    root.reloadData()
+    viewController.dataSource.intervals = intervals
+    viewController.collectionView.reloadData()
   }
 
   override var layoutModifiers: LayoutModifier = LayoutModifier
 
-  override val value: UIView get() = root
+  override val value: UIView get() = viewController.collectionView
+}
+
+private class CollectionViewController<A : AppService>(
+  treehouseApp: TreehouseApp<A>,
+  widgetSystem: WidgetSystem,
+  layout: UICollectionViewLayout,
+) : UICollectionViewController(layout),
+  UICollectionViewDelegateFlowLayoutProtocol {
+  val dataSource = CollectionViewDataSource(treehouseApp, widgetSystem)
+
+  init {
+    collectionView.registerClass(CollectionViewCell, "CollectionViewCell")
+    collectionView.dataSource = dataSource
+  }
+
+  override fun collectionView(collectionView: UICollectionView, layout: UICollectionViewLayout, sizeForItemAtIndexPath: NSIndexPath): CValue<CGSize> {
```

Comment:
CI and Android Studio complain about this method, yet Xcode is fine with it:

```
e: file:///Users/runner/work/redwood/redwood/redwood-treehouse-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/treehouse/lazylayout/uiview/UIViewLazyList.kt:74:9 Class 'CollectionViewController' must override public open expect fun collectionView(collectionView: UICollectionView, layout: UICollectionViewLayout, referenceSizeForHeaderInSection: NSInteger /* = Long */): CValue<CGSize> defined in platform.UIKit.UICollectionViewDelegateFlowLayoutProtocol because it inherits multiple interface methods of it
> Task :redwood-treehouse-lazylayout-uiview:compileCommonMainKotlinMetadata FAILED
e: file:///Users/runner/work/redwood/redwood/redwood-treehouse-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/treehouse/lazylayout/uiview/UIViewLazyList.kt:74:9 'public open expect fun collectionView(collectionView: UICollectionView, layout: UICollectionViewLayout, referenceSizeForHeaderInSection: NSInteger /* = Long */): CValue<CGSize> defined in app.cash.redwood.treehouse.lazylayout.uiview.CollectionViewController' clashes with 'public open expect fun collectionView(collectionView: UICollectionView, layout: UICollectionViewLayout, insetForSectionAtIndex: NSInteger /* = Long */): CValue<UIEdgeInsets> defined in platform.UIKit.UICollectionViewDelegateFlowLayoutProtocol': return types are incompatible
```

Reply:
That's odd--looks like the compiler is confusing this with different methods that have the same "name", but different parameters (i.e. `referenceSizeForHeaderInSection` or `insetForSectionAtIndex`, instead of `sizeForItemAtIndexPath`).

Reply:
That's really weird. All of these methods are expected to be overridden for the `UICollectionViewDelegateFlowLayoutProtocol`

also i think you need to set `layout.delegate = self`